### ISSUE TITLE
case_sensitive_file_extension_comparisons should suggest `is_some_and`

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#14357]
### Summary

The `clippy::case_sensitive_file_extension_comparisons` suggestion should not cause subsequent clippy lint.

### Reproducer

I tried this code:

```rust
fn foo(v: &str)->bool {
    v.ends_with(".a")
}

fn main() {
    println!("test = {:?}", foo("test.a"));
}
```

I expected the suggestion to not cause any clippy warnings, but instead, it suggested to replace it with this:

```rust
2 ~     std::path::Path::new(v)
3 +         .extension()
4 +         .map_or(false, |ext| ext.eq_ignore_ascii_case("a"))
 ```

The suggested code produces this:

```rust
  = note: `#[warn(clippy::unnecessary_map_or)]` on by default
help: use is_some_and instead
  |
2 ~     std::path::Path::new(v)
3 +         .extension().is_some_and(|ext| ext.eq_ignore_ascii_case("a"))
  |
```

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17)
binary: rustc
commit-hash: 4d91de4e48198da2e33413efdcd9cd2cc0c46688
commit-date: 2025-02-17
host: x86_64-unknown-linux-gnu
release: 1.85.0
LLVM version: 19.1.7
```
